x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.test.ts

```diff
@@ -0,0 +1,1026 @@
+      });
+    });
+  });
+
+  describe('hideCredentialsForm', () => {
+    describe('activeApiTokenRawName', () => {
+      it('resets this value', () => {
+        mount({
+          activeApiTokenRawName: 'foo',
+        });
+        CredentialsLogic.actions.hideCredentialsForm();
+        expect(CredentialsLogic.values.activeApiTokenRawName).toEqual('');
+      });
+    });
+    describe('showCredentialsForm', () => {
+      it('resets this value', () => {
+        mount({
+          showCredentialsForm: true,
+        });
+        CredentialsLogic.actions.hideCredentialsForm();
+        expect(CredentialsLogic.values.showCredentialsForm).toEqual(false);
+      });
+    });
+    it('should not change any other values', () => {
+      mount();
+      CredentialsLogic.actions.hideCredentialsForm();
+      expect(CredentialsLogic.values).toEqual({
+        ...DEFAULT_VALUES,
+        showCredentialsForm: expect.any(Boolean),
+        activeApiTokenRawName: expect.any(String),
+      });
+    });
+  });
+
+  describe('resetCredentials', () => {
+    describe('isCredentialsDetailsComplete', () => {
+      it('should reset to false', () => {
+        mount({
+          isCredentialsDetailsComplete: true,
+        });
+        CredentialsLogic.actions.resetCredentials();
+        expect(CredentialsLogic.values.isCredentialsDetailsComplete).toEqual(false);
+      });
+    });
+    describe('isCredentialsDataComplete', () => {
+      it('should reset to false', () => {
+        mount({
+          isCredentialsDataComplete: true,
+        });
+        CredentialsLogic.actions.resetCredentials();
+        expect(CredentialsLogic.values.isCredentialsDataComplete).toEqual(false);
+      });
+    });
+    describe('formErrors', () => {
+      it('should reset', () => {
+        mount({
+          formErrors: ['I am an error'],
+        });
+        CredentialsLogic.actions.resetCredentials();
+        expect(CredentialsLogic.values.formErrors).toEqual([]);
+      });
+    });
+    it('should not change any other values', () => {
+      mount();
+      CredentialsLogic.actions.resetCredentials();
+      expect(CredentialsLogic.values).toEqual({
+        ...DEFAULT_VALUES,
+        isCredentialsDetailsComplete: expect.any(Boolean),
+        isCredentialsDataComplete: expect.any(Boolean),
+        formErrors: expect.any(Array),
+      });
+    });
+  });
+
+  describe('initializeCredentialsData', () => {
+    it('should call fetchCredentials and fetchDetails', () => {
+      jest.spyOn(CredentialsLogic.actions, 'fetchCredentials').mockImplementationOnce(() => {});
+      jest.spyOn(CredentialsLogic.actions, 'fetchDetails').mockImplementationOnce(() => {});
+      CredentialsLogic.actions.initializeCredentialsData();
+      expect(CredentialsLogic.actions.fetchCredentials).toHaveBeenCalled();
+      expect(CredentialsLogic.actions.fetchDetails).toHaveBeenCalled();
+    });
+  });
+
+  describe('fetchCredentials', () => {
```

Comment:
FYI: I'm going to stop here for the night + because my requested async/await changes will change the syntax of the below http tests (hopefully to be a bit simpler)? Hope that's cool!